grep/skoved/grep.go

```diff
@@ -0,0 +1,170 @@
+// Package grep provides a Search function for the Grep Exercism exercise
+package grep
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"regexp"
+	"sort"
+)
+
+// grepper provides a Grep style match function
+type grepper struct {
+	lineNums     bool
+	onlyFileName bool
+	ignoreCase   bool
+	invert       bool
+	wholeLine    bool
+	multiFile    bool
+	pattern      string
+}
+
```

Comment:
I like that. I was willing to do the same but i did skip the `struct` lessons on Go - I must go back there.